snuba/web/delete_query.py

```diff
@@ -73,6 +84,33 @@ def delete_from_storage(
 
     results: dict[str, Result] = {}
     attr_info = _get_attribution_info(attribution_info)
+
+    # fail if too many mutations ongoing
+    cluster = storage.get_cluster()
+    if cluster.is_single_node():
+        query = f"""
+    SELECT count() as cnt
+    FROM system.mutations
+    WHERE table IN ({", ".join(map(repr, delete_settings.tables))}) AND is_done=0
+"""
+    else:
+        query = f"""
+SELECT max(cnt)
+FROM (
+    SELECT hostname() as host, count() as cnt
+    FROM clusterAllReplicas('{cluster.get_clickhouse_cluster_name()}', 'system', mutations)
+    WHERE table IN ({", ".join(map(repr, delete_settings.tables))}) AND is_done=0
+    GROUP BY host
+    )
+"""
```

Comment:
Make this piece that determines the amount of mutations its own function. Mock it in a test. then you can test this functionality